plugins/auth-node/src/IdentityClient.ts

```diff
@@ -67,91 +72,33 @@ export class IdentityClient {
     if (!token) {
       throw new AuthenticationError('No token specified');
     }
-    // Get signing key matching token
-    const key = await this.getKey(token);
-    if (!key) {
-      throw new AuthenticationError('No signing key matching token found');
-    }
     // Verify token claims and signature
     // Note: Claims must match those set by TokenFactory when issuing tokens
     // Note: verify throws if verification fails
-    const decoded = JWT.IdToken.verify(token, key, {
+    if (!this.keyStore) {
+      throw new AuthenticationError('No keystore exists');
+    }
+    const decoded = await jwtVerify(token, this.keyStore, {
       algorithms: ['ES256'],
       audience: 'backstage',
       issuer: this.issuer,
-    }) as { sub: string; ent: string[] };
+    });
     // Verified, return the matching user as BackstageIdentity
     // TODO: Settle internal user format/properties
-    if (!decoded.sub) {
+    if (!decoded.payload.sub) {
       throw new AuthenticationError('No user sub found in token');
     }
 
     const user: BackstageIdentityResponse = {
       token,
       identity: {
         type: 'user',
-        userEntityRef: decoded.sub,
-        ownershipEntityRefs: decoded.ent ?? [],
+        userEntityRef: decoded.payload.sub,
+        ownershipEntityRefs: decoded.payload.ent
+          ? [decoded.payload.ent as string]
```

Comment:
Looks like this was `decoded.ent ?? []` previously. Has this changed with the new api?

Reply:
Corrected this to return `decoded.payload.ent` directly and added a test to cover claims decoding